### PR TITLE
[CookBook] Login with Formik

### PR DESCRIPTION
**Pull request by sfernandez11**
## Summary
* Added `withForm` component for wrap the InputFields and inject the Formik props handlers.
* Fixed `lint-diff` script in `package.json`.
* Added `formik` as Form manager.
* Refactored `CustomTextInput`.
* Refactored `Login` screen.
* Refactored authentication flow.
* Added some field validations.

#### Input Validation
gif-inputs-validations

#### Login Success
gif-login-success

#### Login Success
gif-login-error

## Trello Card
https://trello.com/c/f40zHbT9/176-login

### CODE REVIEW

File: generators/app/templates/src/app/screens/Login/index.tsx [hunk 1]
```diff
@@ -1,51 +1,68 @@
...
-        activeOpacity={0.7}
-      />
-    </View>
+  const hasLoginError = useSelector<State, boolean>(store => !!store.auth.currentUserError);
+  const handleLogin: (values: any) => void = useCallback(values => dispatch(AuthActions.login(values)), [
+    dispatch
+  ]);
+  return (
+    <TouchableOpacity activeOpacity={WITHOUT_OPACITY} onPress={Keyboard.dismiss} style={styles.container}>
+      <Formik onSubmit={handleLogin} initialValues={INITIAL_VALUES}>
```
Comment (felire) [inline generators/app/templates/src/app/screens/Login/index.tsx:27]: We should use withFormik here and start avoiding the `Formik` component, what do you think? We could use `yup` to sometimes

File: generators/app/templates/src/app/components/withForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  const { error, touched } = meta;
+  const handleBlur = () => {
+    setIsFocused(false);
+    field.onBlur(name);
+    helper.setTouched(true);
+  };
+  const handleFocus = () => setIsFocused(true);
+  return (
+    <WrappedComponent
+      {...field}
```
[the remark this answers is not in the document]
Comment (sfernandez11) [reply generators/app/templates/src/app/components/withForm/index.tsx:16]: https://jaredpalmer.com/formik/docs/api/useField#fieldinputpropsvalue

File: generators/app/templates/src/app/components/CustomTextInput/index.tsx [hunk 3]
```diff
@@ -1,85 +1,107 @@
...
   multiline: false,
+  placeholder: '',
   returnKeyType: 'done',
   underlineColorAndroid: transparent
 };
 
-export default memo(CustomTextInput);
+const MyCustomTextInput = memo(CustomTextInput);
+
+export const CustomTextInputField = withForm(MyCustomTextInput);
```
[the remark this answers is not in the document]
Comment (sfernandez11) [reply generators/app/templates/src/app/components/CustomTextInput/index.tsx:105]: Done!

File: generators/app/templates/src/app/components/withForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      {...props}
+      error={touched ? error : null}
+      isFocused={isFocused}
+      onBlur={handleBlur}
+      onChange={helper.setValue}
```
[the remark this answers is not in the document]
Comment (sfernandez11) [reply generators/app/templates/src/app/components/withForm/index.tsx:21]: https://jaredpalmer.com/formik/docs/api/useField#fieldhelperprops
`setValue(value: any): void - A function to change the field's value`

File: generators/app/templates/src/app/screens/Login/index.tsx (hunk 1 above)
Comment (sfernandez11) [reply generators/app/templates/src/app/screens/Login/index.tsx:27]: I have my doubts with `withFormik`, maybe in the future but I think it is not necessary for now (here).
`yup` yes 😄 => Nice to have hahaha

**Pull request merged**